pinot-common/src/main/java/org/apache/pinot/common/utils/fetcher/BaseSegmentFetcher.java

```diff
@@ -109,6 +111,46 @@ public File fetchUntarSegmentToLocalStreamed(URI uri, File dest, long rateLimit,
     throw new UnsupportedOperationException();
   }
 
+  /**
+   * @param segmentName
+   * @param uriSupplier the supplier to the list of segment download uris.
+   * @param dest        The destination to put the downloaded segment.
+   * @return true if and only if the segment fetch is successful. This method keeps retrying (with exponential backoff)
+   * of the following steps until segment download is successful or the retry limit is reached whichever comes first 1)
+   * Find servers hosting the segment in ONLINE state from the External View of the table. 2) Shuffle the list of
+   * servers. 3) Go through the list of server http download URIs to fetch the segment until success.
+   * @throws Exception
+   */
+  @Override
+  public boolean fetchSegmentToLocal(String segmentName, Supplier<List<URI>> uriSupplier, File dest)
+      throws Exception {
```

Comment:
We can remove `throws Exception` as we are capturing those in the entire method and returning true / false instead.

Reply:
ditto as my comments above.